addons/api/addon/services/indexed-db.js

```diff
@@ -88,23 +88,46 @@ const unconvertFields = (result, object, schema, serializer) => {
 
 /**
  * Service to encapsulate the IndexedDB implementation. To use this service, call
- * `setup` from the application root and
+ * `setup` from the application root. This will create the database and the necessary indexes.
+ *
+ * Make sure to increment the version number of the database whenever the indexes change.
+ * This will delete the old database and create a new one with the new indexes.
  */
 export default class IndexedDbService extends Service {
   // =attributes
   #db;
+  #version = 1;
+
   get db() {
     return this.#db;
   }
 
-  setup(dbName) {
+  async setup(dbName) {
```

Comment:
Before:
<img width="1783" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/3311ac9c-c70e-46a1-b6dc-f2d23c5d0435">

After:
<img width="1780" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/5783847/3475c700-9726-4b8a-a410-e1adfa43f0d2">

It seems on initial login (not from an authentication restoration) some requests are firing off before index db is properly setup.

I was suspicious that the session service I override [here](https://github.com/hashicorp/boundary-ui/blob/9f11e8a25de5cb89e27b143803d8b5f91755b6da/ui/admin/app/services/session.js#L13) can't actually be async since it seems the [source code](https://ember-simple-auth.com/api/services_session.js.html) doesn't take into account if the overridable method needs to be async. 

The setup call in the service _should_ be awaited but it didn't seem to make a difference. I tried setting up indexed DB in our [authenticate method](https://github.com/hashicorp/boundary-ui/blob/bcf004b8a46af963016a382c738072479dc57d97/ui/admin/app/routes/scopes/scope/authenticate/method.js#L64) before any transition but it seems like for some reason ember still fires off requests for scopes before index DB is ready. 

I'm not sure what the solution is as it seems the moment the `setup` method is async, some of the requests for the ember routes aren't properly awaiting now.